homeassistant/components/minecraft_server/__init__.py

```diff
@@ -127,6 +127,23 @@ async def async_migrate_entry(
 
         _LOGGER.debug("Migration to version 3 successful")
 
+    # 3 --> 4: Remove name in config entry.
+    if config_entry.version == 3:
+        _LOGGER.debug("Migrating from version 3")
+
+        config_data = config_entry.data
+
+        # Migrate config entry.
+        _LOGGER.debug(
+            "Migrating config entry, removing name '%s'", config_data.get("name", "")
+        )
+
+        new_data = config_data.copy()
+        del new_data[CONF_NAME]
+        hass.config_entries.async_update_entry(config_entry, data=new_data, version=4)
+
+        _LOGGER.debug("Migration to version 4 successful")
+
```

Comment:
I think the best option would be not to migrate the entry data and just leave the name there as you don't use it anymore. 
Migration is mostly needed when there is something missing/changed for the new version which you need to fix. The downside of doing migration here is that if a user downgrade HA later the config will break.

Reply:
It's now a little bit weird that I still have to test CONF_NAME in the migration tests. But I definitly see the benefit of compatibility, if some users go back to an older HA version. Much more important. Fixed.